File: 2024/March/3-8 3005 Count Elements With Maximum Frequency/MaxFrequencyCount.py
from typing import List

class Solution:
    def maxFrequencyElements(nums: List[int]) -> int:
        freqMap = {}
        maxFreq = 0
        for num in nums:
            newFreq = freqMap.get(num, 0) + 1
            freqMap[num] = newFreq
            if newFreq > maxFreq:
                maxFreq = newFreq

        count = 0
        for freq in freqMap.values():
            if freq == maxFreq:
                count += freq
        
        return count

nums = [1,2,2,3,1,4]
print(Solution.maxFrequencyElements(nums))

File: 2024/March/3-8 3005 Count Elements With Maximum Frequency/MaxFrequencyCount.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxFrequencyElements = function(nums) {
    const map = new Map();
    var max = 0;
    
    var newFreq;
    for(var num of nums) {
        newFreq = (map.get(num) || 0) + 1;
        if(newFreq > max) {
            max = newFreq;
        }
        map.set(num, newFreq);
    }
    console.log(map);
    var count = 0;
    for(var freq of map.values()) {
        if(freq === max) {
            count += freq;
        }
    }
    
    return count;
};